internal/manifests/osbuild_test.go

```diff
@@ -0,0 +1,544 @@
+
+		osBuildConfig = v1alpha1.OSBuildConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      OSBuildConfigName,
+				Namespace: "a-namespace",
+			},
+			Spec: v1alpha1.OSBuildConfigSpec{
+				Details: v1alpha1.BuildDetails{
+					Distribution: "rhel-86",
+					Customizations: &v1alpha1.Customizations{
+						Packages: []string{"a", "b"},
+						Users:    []v1alpha1.User{{Name: "user-a"}},
+						Services: &v1alpha1.Services{
+							Enabled:  []string{"en-a", "en-b"},
+							Disabled: []string{"dis-1", "dis-2"},
+						},
+					},
+					TargetImage: v1alpha1.TargetImage{
+						Architecture:    "x86_64",
+						TargetImageType: "edge-container",
+						OSTree:          nil,
+					},
+				},
+			},
+			Status: v1alpha1.OSBuildConfigStatus{},
+		}
+
+		expectedOSBuild = v1alpha1.OSBuild{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      configName(OSBuildConfigName, 1),
+				Namespace: osBuildConfig.Namespace,
+			},
+			Spec: v1alpha1.OSBuildSpec{
+				Details:     osBuildConfig.Spec.Details,
+				TriggeredBy: "UpdateCR",
+			},
+		}
+		err := controllerutil.SetControllerReference(&osBuildConfig, &expectedOSBuild, scheme)
+		Expect(err).ToNot(HaveOccurred())
+
+		mockCtrl = gomock.NewController(GinkgoT())
+		osBuildRepository = osbuild.NewMockRepository(mockCtrl)
+		osBuildConfigTemplateRepository = osbuildconfigtemplate.NewMockRepository(mockCtrl)
+		osBuildConfigRepository = osbuildconfig.NewMockRepository(mockCtrl)
+		configMapRepository = configmap.NewMockRepository(mockCtrl)
+
+		creator = manifests.NewOSBuildCRCreator(
+			osBuildConfigRepository,
+			osBuildRepository,
+			scheme,
+			osBuildConfigTemplateRepository,
+			configMapRepository,
+		)
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+	})
+
+	Context("without template", func() {
+		It("should create first OSBuild", func() {
+			// given
+			cp := osBuildConfig.DeepCopy()
+			one := 1
+			cp.Status.LastVersion = &one
+			osBuildConfigRepository.EXPECT().PatchStatus(ctx, cp, gomock.Any())
+
+			osBuildRepository.EXPECT().Create(ctx, &expectedOSBuild)
+
+			// when
+			err := creator.Create(ctx, &osBuildConfig)
+
+			//then
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should create subsequent OSBuild", func() {
+			// given
+			previousVersion := 10
+			osBuildConfig.Status.LastVersion = &previousVersion
+
+			cp := osBuildConfig.DeepCopy()
+			nextVersion := 11
+			cp.Status.LastVersion = &nextVersion
+			osBuildConfigRepository.EXPECT().PatchStatus(ctx, matchers.NewOSBuildConfigStatusMatcher(cp), gomock.Any())
+
+			expectedOSBuild.Name = configName(OSBuildConfigName, nextVersion)
+
+			osBuildRepository.EXPECT().Create(ctx, &expectedOSBuild)
+
+			// when
+			err := creator.Create(ctx, &osBuildConfig)
+
+			//then
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should fail on OSBuildConfig patch failure", func() {
+			// given
+			osBuildConfigRepository.EXPECT().PatchStatus(ctx, gomock.Any(), gomock.Any()).Return(fmt.Errorf("boom"))
```

Comment:
I prefer to mention the specific parameter instead of `gomock.Any` (the second parameter - osbuildConfig)

Reply:
Done.